wake/build.wake

```diff
@@ -59,3 +59,30 @@ global def e300ArtyDUTPlan =
     Nil
   makeRocketChipDUTPlan name e300ScalaModule testharness ""
   | setRocketChipDUTPlanConfigs configs
+
+def devKitFrequencyConfig = match _
+  25  = "sifive.freedom.unleashed.WithDevKit25MHz"
+  50  = "sifive.freedom.unleashed.WithDevKit50MHz"
+  100 = "sifive.freedom.unleashed.WithDevKit100MHz"
+  125 = "sifive.freedom.unleashed.WithDevKit125MHz"
+  150 = "sifive.freedom.unleashed.WithDevKit150MHz"
+  200 = "sifive.freedom.unleashed.WithDevKit200MHz"
+  frequency =
+    def _ = printLevel logError "FPGADevKit: {str frequency} is not a valid frequency"
+    "INVALID.FREQUENCY.CONFIG"
+
+global def makeVC707DUTPlan name frequency designConfigs designScalaModules =
+  def testharness = "sifive.fpgashells.shell.xilinx.VC707Shell"
+  def configs =
+    def frequencyConfig = devKitFrequencyConfig frequency
+    designConfigs ++ (frequencyConfig, Nil)
+  def scalaModules =
+    designScalaModules ++ (u500ScalaModule, Nil)
+  makeRocketChipDUTPlan name u500ScalaModule testharness ""
+  | setRocketChipDUTPlanScalaModules scalaModules
+  | setRocketChipDUTPlanConfigs configs
+
+global def makeVCU118DUTPlan name frequency designConfigs designScalaModules =
+  def testharness = "sifive.fpgashells.shell.xilinx.VCU118Shell"
+  makeVC707DUTPlan name frequency designConfigs designScalaModules
```

Comment:
This seems like an incorrect nesting of dependencies, since Xilinx's VCU118 board is not a "superclass" of a VC707 board. I think a better separation would be to pull out the things that are in common between the two into a separate (private?) function, and then to have both depend on that rather than having the VCU118 depend on the VC707.

In fact, if the only thing that is difference is the Scala path to the shell, then perhaps the `makeVC707DUTPlan` and the `makeVCU118DUTPlan` should be the same function, and then the caller just sets the shell path.

Reply:
sounds good to me. I'll replace these with a single function.